test/unit/src/common/selectors/t_selectors.js

```diff
@@ -0,0 +1,37 @@
+import expect from 'expect';
+
+import { hasNoRegisteredNames } from '../../../../../src/common/selectors';
+
+describe('hasNoRegisteredNames', function() {
+
+  const stateWithNoName = {
+    snaps: {
+      snaps: [{}, {}]
+    }
+  };
+
+  const stateWithName = {
+    snaps: {
+      snaps: [{
+        store_name: 'bsi-test-ii'
+      }, {}]
+    }
+  };
+
+  it('should be false when no names in store', function() {
+    expect(hasNoRegisteredNames(stateWithNoName)).toBe(true);
+  });
+
+  it('should be true when name in snaps', function() {
+    expect(hasNoRegisteredNames(stateWithName)).toBe(false);
+  });
+
+  it('should not memoize the selector', function() {
+    expect(hasNoRegisteredNames.recomputations()).toBe(2);
```

Comment:
See https://github.com/reactjs/reselect#q-how-do-i-test-a-selector

This is the number of times it recomputed the selector, and it should have done it twice because we changed state twice, next test doesn't change state so we get the memoized value.